nautobot_ssot/contrib/adapter.py

```diff
@@ -58,6 +68,38 @@ def invalidate_cache(self, zero_out_hits=True):
         if zero_out_hits:
             self._cache_hits = defaultdict(int)
 
+    def bulk_create(self, model_class, models):
+        with transaction.atomic():
+            for model in models:
+                signals.pre_save.send(sender=model_class, instance=model)
+
+            model_class.objects.bulk_create(models)
+            for model in models:
+                signals.post_save.send(sender=model_class, instance=model)
```

Comment:
Feels like a fully-equivalent approach would be to also call `model.full_clean()` for each model instance. I presume skipping that is where the time savings is mostly coming from?

Reply:
I did some testing with calling `clean` for each object and the performance was still significantly slow. I'll try it again for a clearer benchmark.

Reply:
Okay, I added a `full_clean` (see commit 6c63a921cf18bf10d7e4eb6591cee0e853fc84c5). With this change the >80k location create takes just over 16 hours. So definitely an improvement over the 72 hour run, but still very slow.

Reply:
80% improvement is nothing to sneeze at, but I understand your concern! It would be interesting to get a profile of the 16-hour run and see where full_clean() is spending so much time.